Lusas_Adapter/CRUD/Read/Results/BarResults.cs

```diff
@@ -90,10 +98,56 @@ private IEnumerable<IResult> ExtractBarForce(List<int> ids, List<int> loadcaseId
 
             IFView view = m_LusasApplication.getCurrentView();
             IFResultsContext resultsContext = m_LusasApplication.newResultsContext(view);
-
-            string entity = "Force/Moment - Thick 3D Beam";
+            string entityBeam = "Force/Moment - Thick 3D Beam";
+            string entityBar = "Force/Moment - Bar";
             string location = "Feature extreme";
 
+            // Extract the BarType for all ids provided
+            // List ids for beams, and ids for bars
+            List<int> barIds = new List<int>();
+            List<int> beamIds = new List<int>();
+            foreach (int id in ids)
+            {
+                IFLine lusasLine = d_LusasData.getLineByNumber(id);
+
+                object[] meshAssignments = lusasLine.getAssignments("Mesh");
+
+                if (meshAssignments.Length > 0)
+                {
+                    foreach (object assignment in meshAssignments)
+                    {
+                        IFAssignment lusasAssignment = (IFAssignment)assignment;
+                        IFAttribute lusasMesh = lusasAssignment.getAttribute();
+                        IFMeshLine lusasLineMesh = (IFMeshLine)lusasMesh;
+
+                        object[] barMeshName = lusasLineMesh.getElementNames();
+                        int id2 = int.Parse(lusasLineMesh.getID().ToString());
+                        foreach (object type in barMeshName)
+                        {
+                            if (type.ToString() == "BMI21" ||
+                                type.ToString() == "BMI31" ||
+                                type.ToString() == "BMX21" ||
+                                type.ToString() == "BMX31" ||
+                                type.ToString() == "BMI21W" ||
+                                type.ToString() == "BMI31W" ||
+                                type.ToString() == "BMX21W" ||
+                                type.ToString() == "BMX31W")
+                                beamIds.Add(id);
+                            else if (
+                                type.ToString() == "BRS2" ||
+                                type.ToString() == "BRS3")
+                                barIds.Add(id);
+                        }
+                    }
+                }
+            }
+
+            if (beamIds.Count() + barIds.Count() == 0)
+            {
+                BH.Engine.Base.Compute.RecordWarning("The element type does not match the types currently supported, the adapter will attempt to pull results for a Thick 3D Beam");
+                 beamIds = ids;
+            }
```

Comment:
Surely here, if the count of both bars are 0, we just don't proceed with the rest of the method?

What we could do is have an if statement where we put the `GetFeatureResults` method and retrieval of values is put in a private method and then called if the count of Ids > 0.